juju/client/connector.py

```diff
@@ -0,0 +1,121 @@
+import asyncio
+import logging
+import macaroonbakery.httpbakery as httpbakery
+
+from juju.client.connection import Connection
+from juju.client.jujudata import JujuData
+from juju.errors import JujuConnectionError, JujuError
+
+log = logging.getLogger('connector')
+
+
+class Connector:
+    def __init__(self, loop=None, max_frame_size=None, bakery_client=None):
+        self.max_frame_size = max_frame_size
+        self.loop = loop or asyncio.get_event_loop()
+        self.bakery_client = bakery_client
+        self._connection = None
+        self.controller_name = None
+        self.model_name = None
+        self.jujudata = JujuData()
+
+    def is_connected(self):
+        return self._connection is not None
+
+    def connection(self):
+        if not self.is_connected():
+            raise Exception('not connected')
```

Comment:
Please raise a subclass of Exception so that callers can target it specifically.

Reply:
If you want to avoid the exception, you can call is_connected. I'm not entirely convinced it's worth it, but done anyway.